### PR TITLE
#154 Add login integration with LeanMoodBackend 

### LINKED ISSUE #154: Add login integration with LeanMoodBackend

### PR DESCRIPTION
**Pull request by Nasdan**
Close #154

### CODE REVIEW

File: config/webpack/app/webpack.config.base.js [hunk 1]
```diff
@@ -97,6 +97,7 @@ module.exports = merge(common, {
     }),
     new webpack.DefinePlugin({
       'process.env.LM_REST_ENV': JSON.stringify(env.LM_REST_ENV),
+      'process.env.API_URL': JSON.stringify(env.API_URL),
```
Comment (brauliodiez) [inline config/webpack/app/webpack.config.base.js]: BASE_API_URL?

File: src/rest-api/login/loginAPI.real.ts [hunk 2]
```diff
@@ -2,17 +2,39 @@ import { LoginCredentials } from '../../model/login/loginCredentials';
 import { LoginResponse } from '../../model/login/loginResponse';
 import { loginMockResponses } from './loginMockData';
 import { LoginFunction } from './loginAPI.contract';
+import { User } from '../model/general';
+import { mapUserToModel } from '../mappers/general';
```
Comment (brauliodiez) [inline src/rest-api/login/loginAPI.real.ts]: MapUserToStateModel?

File: config/webpack/app/webpack.config.base.js (hunk 1 above)
Comment (Nasdan) [inline config/webpack/app/webpack.config.base.js]: Done

File: src/rest-api/login/loginAPI.real.ts (hunk 2 above)
Comment (Nasdan) [inline src/rest-api/login/loginAPI.real.ts]: Done

**Pull request merged**